agent/dehydrate.js

```diff
@@ -77,8 +77,10 @@ function dehydrate(data: Object, cleaned: Array<Array<string>>, path?: Array<str
   // TODO when this is in the iframe window, we can just use Object
   if (data.constructor && typeof data.constructor === 'function' && data.constructor.name !== 'Object') {
     cleaned.push(path);
+    const isDate = data.constructor.name === 'Date';
```

Comment:
Is this check guaranteed to work? Maybe we can do something safer, like `Object.prototype.toString.call(data)`?